File: src/app/lib/models/events_response.dart
import 'event_dto.dart';

/// API response envelope wrapping event data.
class EventsResponse {
  final List<EventDto> events;
  final DateTime lastUpdated;
  final bool cacheHit;

  const EventsResponse({
    required this.events,
    required this.lastUpdated,
    required this.cacheHit,
  });

  factory EventsResponse.fromJson(Map<String, dynamic> json) =>
      EventsResponse(
        events: (json['events'] as List<dynamic>)
            .map((e) => EventDto.fromJson(e as Map<String, dynamic>))
            .toList(),
        lastUpdated: DateTime.parse(json['lastUpdated'] as String),
        cacheHit: json['cacheHit'] as bool,
      );

  Map<String, dynamic> toJson() => {
        'events': events.map((e) => e.toJson()).toList(),
        'lastUpdated': lastUpdated.toIso8601String(),
        'cacheHit': cacheHit,
      };
}

File: src/backend/GoCalGo.Contracts/Events/EventsResponse.cs
using System.Text.Json.Serialization;

namespace GoCalGo.Contracts.Events
{
    /// <summary>
    /// API response envelope wrapping event data.
    /// </summary>
    public sealed record EventsResponse
    {
        /// <summary>List of events matching the query.</summary>
        [JsonPropertyName("events")]
        public required IReadOnlyList<EventDto> Events { get; init; }

        /// <summary>Timestamp of the last successful data ingestion from ScrapedDuck.</summary>
        [JsonPropertyName("lastUpdated")]
        public required DateTime LastUpdated { get; init; }

        /// <summary>Whether this response was served from cache.</summary>
        [JsonPropertyName("cacheHit")]
        public required bool CacheHit { get; init; }
    }
}
